web/crux/src/app/deploy/deploy.service.ts

```diff
@@ -555,6 +549,71 @@ export default class DeployService {
     return CreateEntityResponse.fromJSON(newDeployment)
   }
 
+  async preDeploymentCheck(id: string): Promise<Empty> {
+    const deployment = await this.prisma.deployment.findFirstOrThrow({
+      where: {
+        id,
+      },
+    })
+
+    const node = this.agentService.getById(deployment.nodeId)
+    if (!node || node.getConnectionStatus() !== NodeConnectionStatus.CONNECTED) {
+      throw new PreconditionFailedException({
+        message: 'Node is unreachable',
+        property: 'nodeId',
+        value: deployment.nodeId,
+      })
+    }
+
+    const secretsHaveValue = await this.requiredSecretsHaveValue(id)
+    if (!secretsHaveValue) {
+      throw new PreconditionFailedException({
+        message: 'Required secrets must have values!',
+        property: 'deploymentId',
+        value: deployment.id,
+      })
+    }
+
+    return {}
```

Comment:
Is this really needed?

Reply:
Originally I used this function for a gRPC request and I forgot this here :)